evap/staff/forms.py

```diff
@@ -176,6 +176,18 @@ def __init__(self, *args, **kwargs):
             # form is used as read-only course view
             disable_all_fields(self)
 
+    def validate_unique(self):
+        super().validate_unique()
+        # name_xy and semester are unique together. This will be treated as a non-field-error since two
+        # fields are involved. Since we only show the name_xy field to the user, assign that error to this
+        # field.
+        for e in self.non_field_errors().as_data():
+            if e.code == "unique_together" and "unique_check" in e.params:
+                if "semester" in e.params["unique_check"]:
+                    field = e.params["unique_check"][0]
+                    field = e.params["unique_check"][1] if field == "semester" else field
```

Comment:
these two lines are rather unclear. i would suggest to rename `field` to `name_field` and actually i would suggest to simply access the second field and be done with it.